pkg/apply/task/inv_set_task.go

```diff
@@ -152,24 +154,44 @@ func (i *DeleteOrUpdateInvTask) updateInventory(taskContext *taskrunner.TaskCont
 
 	// If an object is invalid and was previously stored in the inventory,
 	// then keep it in the inventory so it can be applied/pruned next time.
-	invalidObjects := i.PrevInventory.Intersection(taskContext.InvalidObjects())
+	invalidObjects := prevInventory.Intersection(taskContext.InvalidObjects())
 	klog.V(4).Infof("keep in inventory %d invalid objects", len(invalidObjects))
 	invObjs = invObjs.Union(invalidObjects)
 
 	klog.V(4).Infof("get the apply status for %d objects", len(invObjs))
-	objStatus := taskContext.InventoryManager().Inventory().Status.Objects
+	objStatus := taskContext.InventoryManager().Inventory().ObjStatuses
 
 	klog.V(4).Infof("set inventory %d total objects", len(invObjs))
-	err := i.InvClient.Replace(taskContext.Context(), i.InvInfo, invObjs, objStatus, i.DryRun)
+	// Skip entire function for dry-run.
```

Comment:
Could this be done at the top of the function?

Or should we maybe compute the new inventory and log the diff when doing a dry-run?

Reply:
Maybe just clarify the comment then "Exit before updating the inventory, but after logging the above changes"